virtualizarr/manifests/utils.py

```diff
@@ -157,6 +157,21 @@ def _remove_element_at_position(t: tuple[int, ...], pos: int) -> tuple[int, ...]
     return tuple(new_l)
 
 
+def check_no_partial_chunks_on_concat_axis(
```

Comment:
The place to test this is here:

https://github.com/zarr-developers/VirtualiZarr/blob/627a9049e9eb0f2e69a40f3d511ce7ebac3ee45d/virtualizarr/tests/test_manifests/test_array.py#L352

because we should be able to test this error message without ever doing IO.